### PR TITLE
[FIX] stock: multi-steps, backorders and return

### PR DESCRIPTION
**Pull request by nim-odoo**

### CODE REVIEW

**Comment by nim-odoo** [comment]
fw-bot up to saas-12.3

**Comment by Swapnesh-SerpentCS** [comment]
This should fix https://github.com/odoo/odoo/issues/52459 
(or maybe OPW number is wrong in the PR description :thinking: )

File: addons/stock/wizard/stock_picking_return.py [hunk 1]
```diff
@@ -137,8 +137,14 @@ def _create_returns(self):
                 # |       return pick(Add as dest)          return toLink                    return ship(Add as orig)
                 # +--------------------------------------------------------------------------------------------------------+
                 move_orig_to_link = return_line.move_id.move_dest_ids.mapped('returned_move_ids')
+                # link to original move
+                move_orig_to_link |= return_line.move_id
+                # link to siblings of original move, if any
+                move_orig_to_link |= return_line.move_id\
+                    .move_dest_ids.filtered(lambda m: m.state not in ('cancel'))\
```
Comment (fmdl) [inline addons/stock/wizard/stock_picking_return.py:144]: `lambda m: m.state != 'cancel'`

**Comment by nim-odoo** [comment]
fw-bot up to master

**Comment by nim-odoo** [comment]
robodoo rebase-ff

**Comment by nim-odoo** [comment]
@sle-odoo @amoyaux Sorry to bother you guys, but this quite is a bit cumbersome... Your input is welcome.

File: addons/stock/wizard/stock_picking_return.py [hunk 2]
```diff
@@ -141,8 +141,8 @@ def _create_returns(self):
                 move_orig_to_link |= return_line.move_id
                 # link to siblings of original move, if any
                 move_orig_to_link |= return_line.move_id\
-                    .move_dest_ids.filtered(lambda m: m.state not in ('cancel'))\
-                    .move_orig_ids.filtered(lambda m: m.state not in ('cancel'))
+                    .mapped('move_dest_ids.move_orig_ids')\
```
Comment (amoyaux) [inline addons/stock/wizard/stock_picking_return.py:144]: @nim-odoo are you sure, you don't need the intermediate filter anymore?

Comment (nim-odoo) [reply addons/stock/wizard/stock_picking_return.py:144]: I can still keep it, just in case

**Comment by nim-odoo** [comment]
robodoo r+

**Comment by nim-odoo** [comment]
robodoo retry

**Pull request closed without merging**